### ISSUE TITLE
TYPO3 dependency in composer.json is improvable

### ISSUE THREAD

**Issue by h-e-l-l-o-w-o-r-l-d** [bithost-gmbh/pdfviewhelpers#196]
Hi,

I see several problems with your json dependency:

- As 2.4 supports TYPO3 11, the depency must be <=11.5.99 instad of <11.5.99. Because of this our self written tool doesn't detect the extension as TYPO3 11 ready.
- It's probably bad practise to mixup versions with and without "v" as prefix. So it should be 9.5.0 instead of v9.5.0. Almost all extensions have NO v-prefixes.
-  Almost all extension separate dependencies with "||" or ",". In your case there's just a space.

To sum it up:

Now: "typo3/cms-core": ">=v9.5.0 <11.5.99",
Should be: "typo3/cms-core": ">=9.5.0, <=11.5.99",

**Comment by maechler** [comment]
Hi @h-e-l-l-o-w-o-r-l-d 

I agree that the version constraints could be improved, I disagree however that they are currently wrong. Your tool should be capable of reading such a version constraint (without comma and with <) because it is perfectly valid. 
Also see:  https://getcomposer.org/doc/articles/versions.md

Changing it to `<=11.5.99` does not make a whole lot of a difference, it just additionally allows for `11.5.99`, but would still fail for `11.5.100`. Thus I would recommend changing it to `^9.5 || ^10.4 || ^11.5`.

**Comment by h-e-l-l-o-w-o-r-l-d** [comment]
You are of course right, sorry. The tool is fine without comma, just the < needs some tweaking.

However, it's a bit off from other extensions. And <= instead of < is more logical, because the version you see is in the range then and that is easier to grasp. ;)

Your suggestion with || is probably the best anyway, so go for it.
